File: Busca/Binaria/BuscaBinaria.java
package algoritmos;

public class BuscaBinaria {

	public static void main(String[] args) {
		int[] dados = { 1, 2, 30, 41, 73, 81, 90, 101 };

		System.out.println(BuscaBin(dados, 81));
		System.out.println(BuscaBin(dados, 103));
	}

	public static int BuscaBin(int[] lista, int valorBuscado) {
		int cursorBaixo = 0;
		int cursorAlto = lista.length - 1;

		int meio = 0;

		while (cursorBaixo <= cursorAlto) {
			meio = (cursorBaixo + cursorAlto) / 2;
			int chute = lista[meio];

			if (chute == valorBuscado) {
				return meio;
			} else if (chute > valorBuscado) {
				cursorAlto = meio - 1;
			} else {
				cursorBaixo = meio + 1;
			}
		}

		return -1;
	}

	public static void BuscaBinRecursiva() {

	}
}

File: Busca/Binaria/busca_binaria.py
def BuscaBin(lista, item:int) -> int:
    cursor_baixo = 0
    cursor_alto = len(lista) - 1
    
    while (cursor_baixo <= cursor_alto):
        meio = (cursor_baixo + cursor_alto)//2
        chute = lista[meio]
        
        if(chute == item):
            return meio
        if(chute > item):
            cursor_alto = meio - 1    
        else:
            cursor_baixo = meio + 1    
    return -1

######

def BuscaBinRecursiva():
    return -1


dados = [1, 2, 30, 41, 73, 81, 90, 101]

print(BuscaBin(dados, 81))
print(BuscaBin(dados, 103))